codecov.yml

```
@@ -13,4 +13,8 @@ coverage:
         target: 95
         informational: true
     patch: off
-    changes: off
\ No newline at end of file
+    changes: off
+    
+ignore:
+  - "setup.py"
```

You may also want to exclude `tests` as well:

```suggestion
  - "setup.py"
  - "xgcm/tests/*.py"
```

---

Never mind :) Just found out that tests are excluded in `.coveragerc`. How about consolidating codecov config files into one file by moving the contents of `.coveragerc` into `codecov.yml`?

---

Thanks a lot for the suggestion. I have moved the ignore paths over and deleted `.coveragerc`, I did not find equivalents to the other options [in the codecov docs](https://docs.codecov.io/docs/codecovyml-reference), do you have an idea if I need those?

I mean these lines to be specific:
```

[run]
branch = True

[report]
exclude_lines =
    if self.debug:
    pragma: no cover
    raise NotImplementedError
    if __name__ == .__main__.:
ignore_errors = True
```

---

It's my understanding that any line that ends with `# pragma: no cover` is automatically excluded by codecov and as a result you don't need to include it in config file (please correct me if I am wrong or you are getting different behavior).

Regarding `ignore_errors`,  what does it do? I've never had to use it.

---

As far as I can tell it seems to work on codecov. I would go ahead and merge.

---

Sounds good!